internal/cli/daemon/client.go

```diff
@@ -55,13 +57,60 @@ func (c *octaviusClient) startOctaviusClient(grpcClient client.Client) error {
 	return nil
 }
 
+func validateImageName(imageName string) (bool, error) {
+	splitedImageName := strings.Split(imageName, "/")
+	splitedimageTag := strings.Split(imageName, ":")
+	imageNameWithoutTag := splitedimageTag[0]
+	var tag string
+
+	if len(splitedimageTag) == 1 {
+		tag = "latest"
+	} else if len(splitedimageTag) == 2 {
+		tag = splitedimageTag[1]
+	} else {
+		return false, status.Error(codes.Internal, "invalid image tag")
+	}
+
+	var url string
+	if len(splitedImageName) == 1 {
+		url = "https://hub.docker.com/v2/repositories/library/" + imageNameWithoutTag + "/tags/" + tag
+	} else if len(splitedImageName) == 2 {
+		url = "https://hub.docker.com/v2/repositories/" + imageNameWithoutTag + "/tags/" + tag
+	} else {
+		return false, status.Error(codes.Internal, "invalid image name")
+	}
+
+	response, err := http.Get(url)
+	if err != nil {
+		return false, status.Error(codes.Internal, "error in validating image name")
+	}
+	if response.StatusCode == 200 {
```

Comment:
do we get a StatusCode 404 in case when the image is not present in registry? Can we track that and tell the user if image is not found?

Reply:
yes